substrate/frame/election-provider-support/src/lib.rs

```diff
@@ -829,303 +674,10 @@ pub type Voter<AccountId, Bound> = (AccountId, VoteWeight, BoundedVec<AccountId,
-			write!(f, "({:?}, {:?}, {:?}) ", s.0, s.1.total, s.1.voters)?;
-		}
-		Ok(())
-	}
-}
-
-impl<AccountId: PartialEq, BOuter: Get<u32>, BInner: Get<u32>> PartialEq
-	for BoundedSupports<AccountId, BOuter, BInner>
-{
-	fn eq(&self, other: &Self) -> bool {
-		self.0 == other.0
-	}
-}
-
-impl<AccountId, BOuter: Get<u32>, BInner: Get<u32>> Into<Supports<AccountId>>
-	for BoundedSupports<AccountId, BOuter, BInner>
-{
-	fn into(self) -> Supports<AccountId> {
-		// TODO: can be done faster with unsafe code.
-		self.0.into_iter().map(|(acc, b_support)| (acc, b_support.into())).collect()
-	}
-}
-
-impl<AccountId, BOuter: Get<u32>, BInner: Get<u32>>
-	From<BoundedVec<(AccountId, BoundedSupport<AccountId, BInner>), BOuter>>
-	for BoundedSupports<AccountId, BOuter, BInner>
-{
-	fn from(t: BoundedVec<(AccountId, BoundedSupport<AccountId, BInner>), BOuter>) -> Self {
-		Self(t)
-	}
-}
-
-impl<AccountId: Clone, BOuter: Get<u32>, BInner: Get<u32>> Clone
-	for BoundedSupports<AccountId, BOuter, BInner>
-{
-	fn clone(&self) -> Self {
-		Self(self.0.clone())
-	}
-}
-
-impl<AccountId, BOuter: Get<u32>, BInner: Get<u32>> sp_std::ops::Deref
-	for BoundedSupports<AccountId, BOuter, BInner>
-{
-	type Target = BoundedVec<(AccountId, BoundedSupport<AccountId, BInner>), BOuter>;
-
-	fn deref(&self) -> &Self::Target {
-		&self.0
-	}
-}
-
-impl<AccountId, BOuter: Get<u32>, BInner: Get<u32>> IntoIterator
-	for BoundedSupports<AccountId, BOuter, BInner>
-{
-	type Item = (AccountId, BoundedSupport<AccountId, BInner>);
-	type IntoIter = sp_std::vec::IntoIter<Self::Item>;
-
-	fn into_iter(self) -> Self::IntoIter {
-		self.0.into_iter()
-	}
-}
-
-impl<AccountId, BOuter: Get<u32>, BInner: Get<u32>> TryFrom<Supports<AccountId>>
-	for BoundedSupports<AccountId, BOuter, BInner>
-{
-	type Error = crate::Error;
-
-	fn try_from(supports: Supports<AccountId>) -> Result<Self, Self::Error> {
-		// optimization note: pre-allocate outer bounded vec.
-		let mut outer_bounded_supports = BoundedVec::<
-			(AccountId, BoundedSupport<AccountId, BInner>),
-			BOuter,
-		>::with_bounded_capacity(
-			supports.len().min(BOuter::get() as usize)
-		);
-
-		// optimization note: avoid intermediate allocations.
-		supports
-			.into_iter()
-			.map(|(account, support)| (account, support.try_into().map_err(|_| ())))
-			.try_for_each(|(account, maybe_bounded_supports)| {
-				outer_bounded_supports
-					.try_push((account, maybe_bounded_supports?))
-					.map_err(|_| ())
-			})
-			.map_err(|_| crate::Error::BoundsExceeded)?;
-
-		Ok(outer_bounded_supports.into())
-	}
-}
-
-/// Same as `BoundedSupports` but parameterized by an `ElectionProvider`.
+/// Same as `BoundedSupports` but parameterized by a `ElectionProviderBase`.
 pub type BoundedSupportsOf<E> = BoundedSupports<
-	<E as ElectionProvider>::AccountId,
-	<E as ElectionProvider>::MaxWinnersPerPage,
-	<E as ElectionProvider>::MaxBackersPerWinner,
+	<E as ElectionProviderBase>::AccountId,
+	<E as ElectionProviderBase>::MaxWinners,
```

Comment:
Okay, at first I was against this revert.

My idea was that we keep these traits multi-page + capable of being bounded (having things like `MaxWinners` etc), but now I now see that this can cause too much headache to get this revert done.

This will totally fuck me when I merge master, as I do need to bring a lot of it back, but I think it can be useful. 

`pallet-staking` + `pallet-epm` + `frame-ep-support` atm work together, and they are: unbounded, and single page

I will add a new `mod multi_page` here later, which would bring in the traits needed for bounded + multi-page support.

 A bit redundant code, but I can see now that it is a safe choice.

Reply:
but, I think we could have also kept the current version. I don't think there is time for it now anymore though..